WordPress/AbstractThemeSniff.php

```diff
@@ -0,0 +1,1352 @@
+				$this->get_content_width_check( $file_content );
+				$this->get_editor_style_check( $file_content );
+				$this->get_avatar_checks( $file_content );
+				$this->get_custom_menu_check( $file_content );
+				$this->get_post_pagination_check( $file_content );
+				$this->get_post_format_check( $file_content );
+				$this->get_post_thumbnail_check( $file_content );
+				$this->get_post_tags_check( $file_content );
+				$this->get_title_tag_check( $file_content );
+				$this->get_sidebar_checks( $file_content );
+				$this->get_basic_function_checks( $file_content );
+				$this->get_doctype_check( $file_content );
+				$this->get_textdomains( $file_content );
+			}
+			if ( strpos( $themefile , '.css' ) !== false ) {
+				// css files loaded into line array.
+				$file_content = file( $themefile, FILE_IGNORE_NEW_LINES | FILE_SKIP_EMPTY_LINES );
+				if ( trim( $themedir , '\\' ) . '\style.css' === $themefile ) {
+					$this->retrieve_theme_data( $file_content );
+				}
+				$this->get_css_checks( $file_content );
+				$this->get_post_format_css_check( $file_content );
+			}
+		}
+		// Display errors and warnings.
+		$this->get_themeslug();
+		$this->display_errors_and_warnings( $themedir );
+		return $sniff_helper;
+	}
+
+	/**
+	 * This function will go through the files and identify theme supports.
+	 * Verify that an add_theme_support() call is made for any feature the theme
+	 * has been tagged with from the following list: custom-background,
+	 * custom-header, custom-menu, featured-images/post-thumbnails, post-formats,
+	 * custom-logo
+	 * If found set associated theme tags to true.
+	 *
+	 * @param string $file_content contains content of file in a single string.
+	 */
+	public function get_theme_supports( $file_content ) {
+		global $sniff_helper;
+		if ( false !== strpos( $file_content, 'add_theme_support' ) && false !== strpos( $file_content, 'custom-header' ) ) {
+			$sniff_helper['theme_supports']['custom-header'] = true;
+		}
+		if ( false !== strpos( $file_content, 'add_theme_support' ) && false !== strpos( $file_content, 'custom-background' ) ) {
+			$sniff_helper['theme_supports']['custom-background'] = true;
+		}
+		if ( false !== strpos( $file_content, 'add_theme_support' ) && false !== strpos( $file_content, 'custom-logo' ) ) {
+			$sniff_helper['theme_supports']['custom-logo'] = true;
+		}
+		if ( false !== strpos( $file_content, 'add_theme_support' ) && false !== strpos( $file_content, 'post-formats' ) ) {
+			$sniff_helper['theme_supports']['post-formats'] = true;
+		}
+		if ( false !== strpos( $file_content, 'add_theme_support' ) && false !== strpos( $file_content, 'post-thumbnails' ) ) {
+			$sniff_helper['theme_supports']['featured-images'] = true;
+			$sniff_helper['theme_supports']['featured-image-header'] = true;
+		}
+		if ( false !== strpos( $file_content, 'register_nav_menu' ) ) {
+			$sniff_helper['theme_supports']['custom-menu'] = true;
+		}
+		if ( false !== strpos( $file_content, 'wp_nav_menu' ) ) {
```

Comment:
Both functions are needed to so that the theme supports `custom-menu`. The check would be `false !== strpos( $file_content, 'register_nav_menu' ) & false !== strpos( $file_content, 'wp_nav_menu' )`